cwlgen/commandlinetool.py

```diff
@@ -0,0 +1,202 @@
+import logging
+
+# External libraries
+import ruamel.yaml
+
+from .common import CWL_VERSIONS, DEF_VERSION, CWL_SHEBANG, Namespaces, Parameter
+from .requirements import *
+from .utils import literal, literal_presenter, Serializable
+
+logging.basicConfig(level=logging.INFO)
+_LOGGER = logging.getLogger(__name__)
+
+#  Function(s)  ------------------------------
+
+#  Class(es)  ------------------------------
+
+
+class CommandLineTool(Serializable):
+    '''
+    Contain all informations to describe a CWL command line tool.
+    '''
+
+    __CLASS__ = 'CommandLineTool'
+
+    def __init__(self, tool_id=None, base_command=None, label=None, doc=None,
```

Comment:
Yeah I agree, I didn't mainly because I can't default it to a list because it's mutable. I think it's a pretty easy fix though, just default to None and set to empty list of it's None.